### DISCUSSION TITLE
kong ingress controller is not able to find oidc plugin after upgrade to 2.0.x

### DISCUSSION THREAD

**Discussion by suchitsancheti** [revomatico/docker-kong-oidc#28] [Support]
After upgrading the kong to 2.6.0 and kong ingress controller to 2.0.5 and 2.0.6, the ingress controller is giving the following error:

`time="2021-12-07T09:28:24Z" level=error msg="failed to fetch KongPlugin: no KongPlugin or KongClusterPlugin was found" kongplugin_name=oidc kongplugin_namespace=default subsystem=proxy-cache-resolver
time="2021-12-07T09:28:27Z" level=error msg="failed to fetch KongPlugin: no KongPlugin or KongClusterPlugin was found" kongplugin_name=oidc kongplugin_namespace=apps subsystem=proxy-cache-resolver
time="2021-12-07T09:28:27Z" level=error msg="failed to fetch KongPlugin: no KongPlugin or KongClusterPlugin was found" kongplugin_name=oidc kongplugin_namespace=default subsystem=proxy-cache-resolver
`
The configuration for the OIDC plugin is:
`apiVersion: configuration.konghq.com/v1
config:
  client_id: test
  client_secret: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
  discovery: https://keycloak.abc.com/auth/realms/xxxx/.well-known/openid-configuration
  introspection_endpoint: https://keycloak.abc.com/auth/realms/xxxx/protocol/openid-connect/token/introspect
  realm: test
  scope: openid
kind: KongClusterPlugin
metadata:
  name: oidc
plugin: oidc`

The env variable set in of kong:
`KONG_PLUGINS: bundled,oidc`

Please suggest whats wrong here

**Comment by suchitsancheti** [comment]
@cristichiru Can you please suggest the solution for above issue?

**Comment by cristichiru** [comment]
I do not have such problems. Perhaps your KongPlugin is in the wrong namespace (not in default)?

`kubectl get KongPlugin -A`?

**Reply by suchitsancheti** [reply]
Thanks @cristichiru for reverting. I fixed the issue by adding the annotations to the kong cluster plugin as follows:

`kubernetes.io/ingress.class: kong`
